src/main/java/adris/altoclef/tasks/movement/GetToXZWithElytraTask.java

```diff
@@ -0,0 +1,251 @@
+                    _fireWorkTimer.reset();
+                    pitch = (float)-10;
+                }
+            }
+            //Log a message in chat
+            if (_messageProgessTimer.elapsed()) {
+                Debug.logMessage("Distance: "+(int)dist+", Elytra durability: "+durabilityLeft);
+                _messageProgessTimer.reset();
+            }
+        } else { //if the distance is under 15, we need to land slowly
+            setDebugState("Landing...");
+            if (getGroundHeightWithRadius(mod, (int)mod.getPlayer().getPos().x, (int)mod.getPlayer().getPos().z)+50 > mod.getPlayer().getPos().y) {
+                pitch = (float)20; //look a bit down
+            } else {
+                pitch = (float)50; //look down, to land faster
+            }
+        }
+
+        if (dist > 6) { //if the distance is upper than 6
+            if (!LookHelper.isLookingAt(mod, new Rotation(yaw, pitch))) {
+                LookHelper.lookAt(mod, new Rotation(yaw, pitch)); //Look at the target
+            }
+            
+        }
+        //if we have landed, and the distance is under 12 or we don't have any fireworks
+        if (isOnGround(mod) && (dist < 12 || !mod.getItemStorage().hasItem(Items.FIREWORK_ROCKET))) {
+            if (StorageHelper.getItemStackInSlot(new PlayerSlot(6)).getItem() == Items.ELYTRA) { //Unequip elytra
+                return new ClickSlotTask(new PlayerSlot(6)); //Click on the elytra in the armor slot
+            } else if (!StorageHelper.getItemStackInCursorSlot().isEmpty()){ //Once it's in the cursor slot
+                List<Slot> airslot = mod.getItemStorage().getSlotsWithItemPlayerInventory(false, Items.AIR); //Click on a empty inv slot
+                if (airslot.isEmpty()) {
+                    return new EnsureFreeInventorySlotTask(); //If there is no space
+                } else {
+                    return new ClickSlotTask(airslot.get(0)); //Click on the slot to put elytra back in inventory
+                }
+            } else {
+                 _isFlyRunning = false; //Recheck the code at the start of this task to
+            }
+        }
+         _oldCoordsY = mod.getPlayer().getPos().y; //save the old player y position
+        return null;
+    }
+    private boolean isOnGround(AltoClef mod) {
+        return mod.getPlayer().isSwimming() || mod.getPlayer().isTouchingWater() || mod.getPlayer().isOnGround() || mod.getPlayer().isClimbing();
+    }   
+    @Override
+    protected void onStop(AltoClef mod, Task interruptTask) {
+        mod.getBehaviour().disableDefence(false);
+    }
+
+    @Override
+    protected boolean isEqual(Task other) {
+        return other instanceof GetToXZWithElytraTask;
+    }
+
+    @Override
+    public boolean isFinished(AltoClef mod) {
+        return _isFinished;
+    }
+    @Override
+    protected String toDebugString() {
+        return "Moving using Elytra";
+    }
+    private int getGroundHeight(AltoClef mod, int x, int z) {
```

Comment:
it was because `WorldHelper.getGroundHeight` was using "isSolid" instead of "!isAir" to check for the ground height, "isSolid" doesn't includes leaves blocks for idk